src/main/java/org/jvnet/hudson/WireMockRuleFactory.java

```diff
@@ -0,0 +1,60 @@
+package org.jvnet.hudson.test;
+
+import com.github.tomakehurst.wiremock.common.SingleRootFileSource;
+import com.github.tomakehurst.wiremock.core.Options;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+/**
+ * Provides a way to record and mock REST calls.  The underlying structure is 
+ * <a href="http://wiremock.org/">WireMock</a>.  To use: 
+ * <pre>
+ * {@code
+ * WireMockRuleFactory wmrf = new WireMockRuleFactory(api_to_mock);
+ * @Rule
+ * WireMockRule = wmrf.getRule(8089); //port is configurable  
+ * }
+ * </pre>
+ *
+ * Point your system to "http://localhost:8089/" to see mocked responses.
+ *
+ * To record mocks: {@code mvn clean test -DrecordMocks=true}  Clearing 
+ * previously recorded mocks before generating new ones is recommended. 
+ */
+public class WireMockRuleFactory {
+    private String prop = System.getProperty("toRecord");
+    private String urlToMock;
+
+    /**
+     * @param url  The source to mock through this particular WireMock instance
+     */
+    public WireMockRuleFactory(String url){
+        urlToMock = url;
+    } 
+
+    public WireMockRule getRule(int port){
+        return getRule(wireMockConfig().port(port));
+    }
+
+    public WireMockRule getRule(Options options) {
+        if(prop.equalsIgnoreCase("true")) {
+            return new WireMockRecorderRule(options, urlToMock);
+        } else {
+            return new WireMockRule(options);
+        }
+    }
+
+
+    private class WireMockRecorderRule extends WireMockRule {
+        //needed for WireMockRule file location
+        private String mappingLocation = "src/test/resources";
+
+        public WireMockRecorderRule(Options options, String url){
```

Comment:
it would seem that the logical way to do this is have the `wiremock.record` property value be the URL to proxy... if the property is blank then no recording... if the property is non-blank, set up the recorder for the specified url. That makes running easier...